### PR TITLE
docs: update local development guide

### PR DESCRIPTION
**Pull request by majorgreys**
## Checklist
- [ ] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,75 +33,112 @@ Confused about the terminology of APM? Take a look at the [APM Glossary][visuali
...
-    $ riot run psycopg
-
+    
 [tox]: https://github.com/tox-dev/tox/
 [riot]: https://github.com/DataDog/riot/
 
-#### Running Tests locally
+##### Option #2: datadog/dd-trace-py
+
+Use provide a script to start the
```
Comment (gnufede) [inline README.md:100]: Not sure what this sentence means (I'm not a native English speaker).
Maybe reformulate as: "Use the following script..."?

Comment (brettlangdon) [reply README.md:100]:
```suggestion
Use the provided script to start the
```

I bet this is what was intended

File: README.md [hunk 2]
```diff
@@ -114,6 +151,8 @@ runs: `^<TEST_SUITE_NAME>$` where `^` signifies the start of a string and `$`
 signifies the end of a string. For example, use `riot -v run -s -x ^redis$` to
 run only the redis suite.
 
+##### tox
```
Comment (gnufede) [inline README.md:154]: I'm guessing this section is WIP, right?

Comment (brettlangdon) [reply README.md:154]: yes, and hopefully will go away at some point. we only have a few tests still running on `tox`.

File: README.md [hunk 3]
```diff
@@ -33,75 +33,112 @@ Confused about the terminology of APM? Take a look at the [APM Glossary][visuali
...
+    
+We recommend using a virtual environment for local development. 
 
-Once your docker-compose environment is running, you can use the shell script to
-execute tests within a Docker image. You can start the container with a bash shell:
+If you followed the optional step to install the `pyenv-virtualenv` plugin, you can use it to manage your Python virtual environments:
 
-    $ scripts/ddtest
+    pyenv virtualenv 3.9.11 ddtrace-py3.9
+    pyenv activate ddtrace-py3.9
```
Comment (avara1986) [inline README.md:78]: Should we emphasize that py3.9 is the standard to run the tests? if you run `riot` and `mypy` with `riot run -s mypy` it fails in py3.8. I get this error below with autohook:

```
docs/conf.py:286: error: "dict" is not subscriptable, use "typing.Dict" instead  [misc]
docs/conf.py:422: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:422: error: "tuple" is not subscriptable, use "typing.Tuple" instead  [misc]
docs/conf.py:434: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:434: error: "tuple" is not subscriptable, use "typing.Tuple" instead  [misc]
docs/conf.py:473: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:473: error: "tuple" is not subscriptable, use "typing.Tuple" instead  [misc]
docs/conf.py:509: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:530: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:531: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:595: error: "dict" is not subscriptable, use "typing.Dict" instead  [misc]
ddtrace/settings/config.py:22: error: "set" is not subscriptable, use "typing.Set" instead  [misc]
ddtrace/profiling/_asyncio.py:29: error: Unused "type: ignore" comment
ddtrace/profiling/_asyncio.py:38: error: Unused "type: ignore" comment
Found 14 errors in 3 files (checked 389 source files)
```

File: README.md [hunk 4]
```diff
@@ -33,75 +33,112 @@ Confused about the terminology of APM? Take a look at the [APM Glossary][visuali
...
-3. Install the relevant versions of Python in Pyenv: `pyenv install 3.9.1, 2.7.18, 3.5.10, 3.6.12, 3.7.9, 3.8.7, 3.10.0`
-4. Make those versions available globally: `pyenv global 3.9.1, 2.7.18, 3.5.10, 3.6.12, 3.7.9, 3.8.7, 3.10.0`
+The library supports multiple Python runtime versions and platforms. See [supported runtimes](https://ddtrace.readthedocs.io/en/stable/versioning.html#supported-runtimes) for the full list. You have two options for preparing a local development environment. You can use pyenv to install Python runtimes on your local machine or you can use the `datadog/dd-trace-py` Docker image which includes all runtimes.
 
-### Testing
+##### Option #1: pyenv
+
+1. [Install pyenv](https://github.com/pyenv/pyenv#getting-pyenv)
+2. Install Python runtimes with pyenv
+3. [Optional] [Install pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv)
```
Comment (majorgreys) [inline README.md:67]: Can this be removed? The minimum requirement is that a user install riot and tox, which handle creating virtual environments.

File: README.md [hunk 5]
```diff
@@ -33,75 +33,112 @@ Confused about the terminology of APM? Take a look at the [APM Glossary][visuali
...
+
+To install all the tested Python runtimes:
 
-#### Running Tests in docker
+    cd docker; pyenv local | xargs -L 1 pyenv install
```
Comment (majorgreys) [inline README.md:71]: We should symlink the `.python-version` in the root folder to the file in the docker context. Does a user still need to set `pyenv global`?

File: README.md (hunk 3 above)
Comment (majorgreys) [inline README.md:78]:
```suggestion
    pyenv virtualenv 3.9.11 ddtrace
    pyenv activate ddtrace
```

**Pull request closed without merging**